### ISSUE TITLE
merged module and automatic module names

### ISSUE THREAD

**Issue by jperedadnr** [beryx/badass-jlink-plugin#25]
Let's say you have a non-modular dependency, and you have to open your module to it:

```
module my.module {
    requires my.dependency;

    opens my.package to my.dependency;

    exports my.package;
}
```

Building and running works. Even the `jlink` task finish successfully.

However when you run the image you get something like:
```
java.lang.IllegalStateException: Cannot load file my.file
        at my.module.merged.module/my.dependency.package
```

As we haven't opened the module to this new merged module, the error was expected: reflection doesn't work.

Two possible workarounds:

1. Open to all
```
module my.module {
    requires my.dependency;

    opens my.package;

    exports my.package;
}
```

This obviously works, but it is not the safest use of `opens`.

2. Open to the merged module:

```
module my.module {
    requires my.dependency;

    opens my.package to my.dependency, my.module.merged.module;

    exports my.package;
}
```

This produces a warning if you build the project:
```
warning: [module] module not found `my.module.merged.module`
```
But the image runs fine now.

Possible solution: I'm not sure if you can detect that the module was being opened to a non-modular jar that will go into a merged module, and then modify that entry in the module with the new name of the resulting merged module?

**Comment by siordache** [comment]
Fixed in 2.5.1. This release also fixes a similar issue with qualified exports.

**Comment by jperedadnr** [comment]
Cool, thanks, I've tested it successfully.
